WcaOnRails/app/controllers/admin_controller.rb

```diff
@@ -69,11 +70,100 @@ def clear_results_submission
       @competition.update(results_submitted_at: nil)
       flash[:success] = "Results submission cleared."
     else
-      flash[:danger] = "Could not clear the results submission. Maybe results are alredy posted, or there is no submission."
+      flash[:danger] = "Could not clear the results submission. Maybe results are already posted, or there is no submission."
     end
     redirect_to competition_admin_upload_results_edit_path
   end
 
+  def post_results
+    @competition = competition_from_params
+
+    data_tables = {
+      result: Result,
+      scramble: Scramble,
+      inbox_result: InboxResult,
+      inbox_person: InboxPerson,
+      newcomer_person: InboxPerson.where(wcaId: ''),
+      newcomer_result: Result.where("personId REGEXP '^[0-9]+$'"),
+    }
+
+    @existing_data = data_tables.transform_values { |table| table.where(competitionId: @competition.id).count }
+  end
+
+  def import_inbox_results
+    @competition = competition_from_params
+
+    ActiveRecord::Base.transaction do
+      @competition.inbox_results
+                  .joins("LEFT JOIN InboxPersons ON InboxPersons.id = InboxResults.personId AND InboxPersons.competitionId = InboxResults.competitionId")
+                  .select("InboxResults.*, InboxPersons.wcaId AS personWcaId, InboxPersons.countryId AS personCountryIso2")
+                  .each do |inbox_res|
+        person_id = inbox_res.personWcaId.presence || inbox_res.personId
+        person_country = Country.find_by_iso2(inbox_res.personCountryIso2)
+
+        # TODO: This runs a lot of validations and is slow(ish, aka. bearable but could be faster).
+        #   Is it safe to just let Results Team insert rows without Rails validations?
```

Comment:
If I'm not mistaken I think it is: both `InboxResult` and `Result` have similar validations for average, round format and so on, so whatever data is in Inbox, it should be pass the "regular" result validations!

Reply:
From a WST perspective, we just have to change two lines. But I want to be very clear here that I'm leaving this decision entirely to WRT, which also means that you would take the responsibility for any inconsistencies that stem from skipping the validations in this import step.

Personally (not WRT) I'd say double security net at the price of 15% more runtime is better than accidentally introducing unchecked/invalid Result rows into our DB.